students/john_rogers/lesson09/donors.py

```diff
@@ -0,0 +1,130 @@
+"""
+Donor classes for mailroom.py
+Author: JohnR
+Version: 1.0 (Lesson 09)
+Last updated: 2/17/2019
+Notes:
+"""
+
+from datetime import date
+
+
+class Donor(object):
+    """
+    Class for single donor methods and attributes
+    """
+    def __init__(self, first, last, donations=None):
+        self.first = first
+        self.last = last
+        if isinstance(donations, float):
+            donations = [donations]
+        self.donations = list(donations)
+
+    def __repr__(self):
+        return f'{self.first} {self.last} is' \
+            f' listed as a donor.'
+
+    @property
+    def full_name(self):
+        return f'{self.first} {self.last}'
+
+    @property
+    def total_donations(self):
+        return round(sum(self.donations), 2)
+
+    @property
+    def number_of_donations(self):
+        return len(self.donations)
+
+    @property
+    def avg_donation_amount(self):
+        return round(self.total_donations / self.number_of_donations, 2)
+
+    def new_donation(self, amount):
+        self.donations.append(amount)
+
+
+class DonorDataBase(object):
+    """
+    Class for managing multiple donors
+    """
+    today = date.today()
+
+    def __init__(self, donors=None):
+        self.donors = []
+        if donors:
+            self.donors = donors
+
+    def add_donor(self, donor):
+        self.donors.append(donor)
+
+    def donor_names(self):
+        return [d.full_name for d in self.donors]
+
+    def total_amount_raised(self):
+        return sum([sum(d.donations for d in self.donors)])
+
+    def total_number_donations(self):
+        return sum([len(d.donations) for d in self.donors])
```

Comment:
you should use Donor methods here: `[d. number_of_donations for d in self.donors]`